controllers/reconcilers/client_helper.go

```diff
@@ -54,6 +59,26 @@ func (c *ClientHelper) UpdateOwned(ctx context.Context, old, obj client.Object)
 	return nil
 }
 
+func (c *ClientHelper) DidChange() bool {
+	return c.changed
+}
+
+func (c *ClientHelper) IsInProgress() bool {
+	return c.deplInProgress
+}
+
+func (c *ClientHelper) CheckDeploymentInProgress(d *appsv1.Deployment) {
+	if d.Status.AvailableReplicas < d.Status.Replicas {
```

Comment:
What about the case where AvailableReplicas > d.Status.Replicas ? 

```suggestion
	if d.Status.AvailableReplicas != d.Status.Replicas {
```

Reply:
It's debatable but I think we can consider the status is ready/deployed when there are more replicas than necessary (ie. it is being downscaled). Like, it's answering the question "do I have _at least_ the desired scaling ?"
I'm ok to change if you/others think it would be better otherwise.

Reply:
I'm not aware enough on the impact of a pod being removed. Can it generate issues like killing a current connection ?

If yes then we should manage it in the status since it's a quick win. If no let's keep it as is :+1:

Reply:
I guess there could be some data loss. Ok, let's  commit:)